src/aligners/diagonal_transition.rs

```diff
@@ -122,6 +122,194 @@ fn fr_to_pos(d: Fr, f: Fr) -> Pos {
+                }
+                _ => todo!(),
+            }
+        } else {
+            //if there is a match
+            if a[(p.0 - 1) as usize] == b[(p.1 - 1) as usize] {
+                return (Pos(p.0 - 1, p.1 - 1), layer, CigarOp::Match, s as Cost);
+            }
+            //if there is no match
+            else {
+                // Linear Substitution
+                if let Some(cost) = self.cm.sub && f == fronts[s - cost as usize].m()[diagonal] + 2 {
+                        return (
+                            Pos(p.0 - 1, p.1 - 1),
+                            None,
+                            CigarOp::Mismatch,
+                            (s - cost as usize) as Cost,
+                        );
+                    }
+                // Linear Insertion
+                if let Some(cost) = self.cm.ins && f == fronts[s - cost as usize].m()[diagonal + 1] + 1 {
+                        return (
+                            Pos(p.0, p.1 - 1),
+                            None,
+                            CigarOp::Insertion,
+                            (s - cost as usize) as Cost,
+                        );
+                    }
+                // Linear Deletion
+                if let Some(cost) = self.cm.del && f == fronts[s - cost as usize].m()[diagonal - 1] + 1 {
+                        return (
+                            Pos(p.0 - 1, p.1),
+                            None,
+                            CigarOp::Deletion,
+                            (s - cost as usize) as Cost,
+                        );
+                }
+                // Checking affine layers
+                for layer_idx in 0..N {
+                    if f == fronts[s].affine(layer_idx)[diagonal] {
+                        // We cannot return the same position twice, can we?
+                        // Here we will lose CigarOp::AffineClose command for we can't return more than one command per one position. Or we may return explicitly CigarOp::AffineClose instead of what the recursive function gives us.
+                        // The code that implements this suggestion is given commented below the return in this if-statement
+
+                        // return self.parent_position(fronts, a, b, p, Some(layer_idx), s as Cost);
+
+                        // let parent_position_instance = self.parent_position(fronts, a, b, p, Some(layer_idx));
+                        // return (parent_position_instance.0,parent_position_instance.1,CigarOp::AffineClose(layer_idx));
+
+                        return (p, Some(layer_idx), CigarOp::AffineClose(layer_idx), cost);
+                    }
+                }
+            }
+        }
+        unreachable!("Error in parent position function for DT aligner!");
+    }
+
+    fn parent_state(
+        &self,
+        fronts: &Vec<Front<N>>,
+        a: &Sequence,
+        b: &Sequence,
+        d: Fr,
+        cost: Cost,
+        layer: Option<usize>,
+        cigar: &mut Cigar,
```

Comment:
Better to just return `CigarOp` I think

Reply:
The problem may be that in the parent_state function I can return more than one CigarOp, and they maybe different types